newIDE/app/src/Utils/GDevelopServices/Authentication.js

```diff
@@ -141,8 +144,25 @@ export default class Authentication {
 
   getFirebaseUser = (cb: (any, ?FirebaseUser) => void) => {
     if (!this.isAuthenticated()) return cb({ unauthenticated: true });
+    this.auth.currentUser.reload().then(() => cb(null, this.firebaseUser));
```

Comment:
Interesting, why is this useful/needed?

Reply:
When opening the profile, the function `_fetchUserProfile` calls this function to ensure the profile is updated.
However, it gets the Firebase profile from the state, meaning that if the prop `emailVerified` has changed, it will not be propagated. Thus, reload is needed here.

So we can either:

- just call `onRefreshFirebaseProfile` on profile dialog open
- or call the whole `onRefreshUserProfile` but call a `reload()` on the firebase profile.

I suggest the second and I can add a comment explaining why this is needed

Reply:
Fair enough. This method `getFirebaseUser` is only used by `_fetchUserProfile`, so seems ok/low risk!